scene/2d/tile_map_layer.cpp

```diff
@@ -2491,14 +2491,13 @@ HashMap<Vector2i, TileSet::TerrainsPattern> TileMapLayer::terrain_fill_connect(c
 	}
 
 	// Fill the terrains.
-	output = terrain_fill_constraints(can_modify_list, p_terrain_set, constraints);
-	return output;
+	return terrain_fill_constraints(can_modify_list, p_terrain_set, constraints);
 }
 
 HashMap<Vector2i, TileSet::TerrainsPattern> TileMapLayer::terrain_fill_path(const Vector<Vector2i> &p_coords_array, int p_terrain_set, int p_terrain, bool p_ignore_empty_terrains) const {
+	ERR_FAIL_COND_V(tile_set.is_null(), (HashMap<Vector2i, TileSet::TerrainsPattern>()));
+	ERR_FAIL_INDEX_V(p_terrain_set, tile_set->get_terrain_sets_count(), (HashMap<Vector2i, TileSet::TerrainsPattern>()));
```

Comment:
Remember that, when the target type is known, you should be able to simplify the type like:


```
	ERR_FAIL_INDEX_V(p_terrain_set, tile_set->get_terrain_sets_count(), (HashMap()));
```

If you're feeling fancy, you can even use the `{}` syntax:

```
	ERR_FAIL_INDEX_V(p_terrain_set, tile_set->get_terrain_sets_count(), {});
```

Whether this is an improvement is perspective I think, so I'm not sure if I'm suggesting you should change it. I prefer non repetition, but I'll leave it up to you.

Reply:
Good points to consider for sure! To me `HashMap()` looks a bit odd because I don't expect to see generic types without the generics, and `{}` looks bare on its own. I can change it if there is consensus from core, but subjectively I like the PR as it is... though it's a slight preference, I don't mind either way.

Reply:
No need to bring this to Core. This is just a matter of preference, if you prefer it the way it is, that's fine by me.